core/domain/user_query_jobs_one_off.py

```diff
@@ -114,6 +120,39 @@ def _is_maximum_exp_edited_query_satisfied(
             len(user_contributions.edited_exploration_ids) <
             query_model.edited_fewer_than_n_exps)
 
+    @staticmethod
+    def _is_created_collection_query_satisfied(
+            user_settings_model, query_model):
+        """Determines whether a user has created collections."""
+        condition = query_model.created_collection == 'True'
+        user_id = user_settings_model.id
+        collection = collection_models.CollectionRightsModel.query(
+            collection_models.CollectionRightsModel.owner_ids == user_id
+        ).fetch(1)
+        if len(collection) > 0:
+            return True if condition else False
+        return False if condition else True
+
+    @staticmethod
+    def _is_used_logic_proof_interaction_query_satisfied(
+            user_settings_model, query_model):
+        """Determines whether a user has used logic proof
+        interaction in any of the explorations created by the user.
+        """
+        condition = query_model.used_logic_proof_interaction == 'True'
+        user_id = user_settings_model.id
+        user_contributions = user_models.UserContributionsModel.get(user_id)
+        exploration_ids = user_contributions.created_exploration_ids
```

Comment:
Should we maybe use `ExplorationRightsModel` here instead?

Reply:
If I want to use ExplorationRightsModel, I have to query based on the user ids, but fetching the UserContributionsModel using the id, might be faster right?

Reply:
That makes sense.